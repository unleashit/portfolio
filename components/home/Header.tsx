import React from "react";
import Link from "next/link";
import Navigation from "@/components/navigation/navigation";
import { getHomeBlocks } from "../../app/data";
import { findKeyWithValue } from "@/lib/utils.new";
import { CMS } from "@/components/common/cms";
import Hamburger from "@/components/common/hamburger";
// import Hamburger from "@/components/common/hamburger";

export default async function Header() {
  const blocks = await getHomeBlocks();
  const fEE = findKeyWithValue(blocks, "title", "Front End Engineer");

  return (
    <header className="main-header">
      <Hamburger />
      {/* @ts-expect-error Server Component */}
      <Navigation />
      <div className="title-tagline-wrapper">
        <h1>{fEE.title}</h1>
        <CMS html={fEE.content} name={"front end engineer"} />
      </div>
      <div className="button-wrapper">
        <Link href="/#work">
          <button className="button button-green">See my work</button>
        </Link>
      </div>
    </header>
  );
}
