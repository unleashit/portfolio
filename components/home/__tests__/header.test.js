import Header from '../header';

describe('<Header />', () => {
    let wrapper;
    let props = {};

    beforeEach(() => {
        wrapper = shallow(<Header {...props} />);
    });

    test('renders without crashing', () => {
        expect(wrapper.find('header.main-header')).toHaveLength(1);
    });

    test('snapshot matches', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // test('analytics hooks are called', () => {
    //     jest.spyOn(Header.prototype, 'analytics');
    //     const mockReactGA = jest
    //         .spyOn(ReactGA, 'event')
    //         .mockImplementation(() => {});
    //
    //     wrapper = shallow(<Header {...props} />);
    //
    //     const button = wrapper.find('button').first();
    //     button.simulate('click');
    //     expect(wrapper.instance().analytics).toHaveBeenCalledWith(
    //         'See my work'
    //     );
    //
    //     expect(mockReactGA).toHaveBeenCalledTimes(1);
    // });
});
