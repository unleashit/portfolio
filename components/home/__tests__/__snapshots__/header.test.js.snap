// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot matches 1`] = `
<header
  className="main-header"
>
  <Hamburger />
  <Navigation
    home={false}
    logo="2"
    ulClass="main-nav"
  />
  <div
    className="title-tagline-wrapper"
  >
    <h1>
      FRONT END ENGINEER
    </h1>
    <h2>
      20 years of passion for creating modern, engaging full stack user experiences on desktop and mobile. At your service!
    </h2>
  </div>
  <div
    className="button-wrapper"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/#work"
    >
      <button
        className="button button-green"
      >
        See my work
      </button>
    </Link>
  </div>
</header>
`;
